src/Hintman/Hint/NoNewlineAtFileEnd.hs

```diff
@@ -0,0 +1,59 @@
+module Hintman.Hint.NoNewlineAtFileEnd
+       ( getNoNewlineAtFileEndComments
+       ) where
+
+import Hintman.Core.Hint (Hint, HintType (NoNewlineAtFileEnd), Line (..), simpleSuggestion, toLines)
+import Hintman.Core.PrInfo (ModifiedFile (..))
+import Hintman.Core.Review (Comment (..), createComment)
+
+import qualified Data.Text as T
+
+
+-- | Creates 'Comment's on adding newlines in all modified files.
+getNoNewlineAtFileEndComments :: [ModifiedFile] -> [Comment]
+getNoNewlineAtFileEndComments = mapMaybe getFileNoNewlineAtFileEndComment
+
+-- | Takes a file and returns a comment if the last line is not an empty line.
+getFileNoNewlineAtFileEndComment :: ModifiedFile -> Maybe Comment
+getFileNoNewlineAtFileEndComment mf@ModifiedFile{..} = spawnComment $
+    maybeGetNonEmptyLastLine fileLines diffText
+  where
+    diffText :: Text
+    diffText = decodeUtf8 mfContent
+
+    fileLines :: [Line]
+    fileLines = toLines diffText
+
+    hint :: Hint
+    hint = simpleSuggestion NoNewlineAtFileEnd "No newline at end of file" ""
+
+    -- | Spawns a "NoNewlineAtFileEnd" comment on a given line.
+    spawnComment :: Maybe Line -> Maybe Comment
+    spawnComment = (=<<) (\l -> createComment mf l hint)
+
+-- | Checks if file ends with a newline, and if it does not, spawns a warning on the last line.
+maybeGetNonEmptyLastLine :: [Line] -> Text -> Maybe Line
+maybeGetNonEmptyLastLine mfLines fileContent = do
+    -- 1. Get the last line of the file.
+    lastLine <- viaNonEmpty last mfLines
+
+    -- 2. Return line for warning creation if the last symbol isn't '\n'.
+    maybeLineForWarning lastLine lastSymbol
+  where
+    {- | Checks whether a symbol can be considered a newline.
+    'Nothing' is considered a newline as well, since this means an empty file. -}
+    isNewline :: Maybe Char -> Bool
+    isNewline ls = case ls of
+        Nothing     -> True
+        Just symbol -> symbol == '\n'
+
+    -- | Takes a last symbol of file (if there is any).
+    lastSymbol :: Maybe Char
+    lastSymbol = if T.null fileContent then Nothing
```

Comment:
I have a strong feeling that I've reinvented the wheel here, but unfortunately my Haskell knowledge and duckduckgo didn't give me the correct answer.

Reply:
There's no such standard function of type `Text -> Maybe Char`, so we can reuse anything here.